tests/test.rfb.js

```diff
@@ -2335,17 +2348,258 @@ describe('Remote Frame Buffer Protocol Client', function () {
+                    let data = [3, 0, 0, 0];
+                    const flags = [0x1F, 0x00, 0x00, 0x01];
+                    let fileSizes = [0x00, 0x00, 0x00, 0x1E];
+
+                    push32(data, utilInt.to_unsigned_32bit(-8));
+                    data = data.concat(flags);
+                    data = data.concat(fileSizes);
+                    client._sock._websocket._receive_data(new Uint8Array(data));
+                });
+
+                afterEach(function () {
+                    RFB.messages.extendedClipboardNotify.restore();
+                });
+
+                it('should send a Notify message with supported formats when receiving a Peek', function () {
+                    let data = [3, 0, 0, 0];
+                    const flags = [0x04, 0x00, 0x00, 0x00];
+                    push32(data, utilInt.to_unsigned_32bit(-4));
+                    data = data.concat(flags);
+                    let expectedData = [0x01];
+
+                    client._sock._websocket._receive_data(new Uint8Array(data));
+
+                    expect(RFB.messages.extendedClipboardNotify).to.have.been.calledOnce;
+                    expect(RFB.messages.extendedClipboardNotify).to.have.been.calledWith(client._sock, expectedData);
+                });
+            });
+
+            describe('Handle Request', function () {
+                beforeEach(function () {
+                    sinon.spy(RFB.messages, 'extendedClipboardProvide');
+
+                    // Send our capabilities
+                    let data = [3, 0, 0, 0];
+                    const flags = [0x1F, 0x00, 0x00, 0x01];
+                    let fileSizes = [0x00, 0x00, 0x00, 0x1E];
+
+                    push32(data, utilInt.to_unsigned_32bit(-8));
+                    data = data.concat(flags);
+                    data = data.concat(fileSizes);
+                    client._sock._websocket._receive_data(new Uint8Array(data));
+                });
+
+                afterEach(function () {
+                    RFB.messages.extendedClipboardProvide.restore();
+                });
+
+                it('should send a Provide message with supported formats when receiving a Request', function () {
+                    let data = [3, 0, 0, 0];
+                    const flags = [0x02, 0x00, 0x00, 0x01];
+                    push32(data, utilInt.to_unsigned_32bit(-4));
+                    data = data.concat(flags);
+                    let expectedData = [0x01];
+
+                    client.clipboardPasteFrom("HejHej");
+
+                    client._sock._websocket._receive_data(new Uint8Array(data));
+
+                    expect(RFB.messages.extendedClipboardProvide).to.have.been.calledOnce;
+                    expect(RFB.messages.extendedClipboardProvide).to.have.been.calledWith(client._sock, expectedData, ["HejHej"]);
+                });
+            });
+        });
+
+        describe('Extended Clipboard Handling Send', function () {
```

Comment:
These are the first tests we have that test `RFB.messages`. I think it might be prudent to create a new grouping that makes this clear.